schema/auth.json

```diff
@@ -0,0 +1,176 @@
+            },
+            {
+              "title": "Bearer Auth Info State",
+              "$ref": "#/definitions/bearerinfodef"
+            },
+            {
+              "title": "Oidc Auth Info",
+              "$ref": "#/definitions/oidcinfodef"
+            }
+          ]
+        }
+      },
+      "required": [
+        "name",
+        "info",
+        "scheme"
+      ]
+    },
+    "basicinfodef": {
+      "oneOf": [
+        {
+          "type": "string",
+          "description": "Expression referencing a workflow secret that contains all needed basic auth info"
+        },
+        {
+          "type": "object",
+          "description": "Basic auth information",
+          "properties": {
+            "username": {
+              "type": "string",
+              "description": "String or a workflow expression. Contains the user name",
+              "minLength": 1
+            },
+            "password": {
+              "type": "string",
+              "description": "String or a workflow expression. Contains the user password",
+              "minLength": 1
+            }
+          },
+          "required": [
+            "username",
+            "password"
+          ],
+          "additionalProperties": false
+        }
+      ]
+    },
+    "bearerinfodef": {
+      "oneOf": [
+        {
+          "type": "string",
+          "description": "Expression referencing a workflow secret that contains all needed bearer auth info"
+        },
+        {
+          "type": "object",
+          "description": "Bearer auth information",
+          "properties": {
+            "token": {
+              "type": "string",
+              "description": "String or a workflow expression. Contains the token",
+              "minLength": 1
+            }
+          },
+          "required": [
+            "token"
+          ],
+          "additionalProperties": false
+        }
+      ]
+    },
+    "oidcinfodef": {
+      "oneOf": [
+        {
+          "type": "string",
+          "description": "Expression referencing a workflow secret that contains all needed oidc auth info"
+        },
+        {
+          "type": "object",
+          "description": "Oidc auth information",
+          "properties": {
+            "grantType": {
+              "type": "string",
+              "description": "Defines the auth type",
+              "enum": [
+                "resourceOwner",
+                "clientCredentials",
+                "tokenExchange"
+              ]
+            },
+            "clientId": {
+              "type": "string",
+              "description": "String or a workflow expression. Contains the client id",
+              "minLength": 1
+            },
+            "clientSecret": {
+              "type": "string",
+              "description": "String or a workflow expression. Contains the client secret",
+              "minLength": 1
+            },
+            "scope": {
```

Comment:
As OIDC allows settings multiple space-separated scopes, maybe we should rename the property `scopes` and turn it into an array of strings. WDYT?

Reply:
updated